src/main/java/org/wso2/carbon/connector/operations/AppendFile.java

```diff
@@ -0,0 +1,131 @@
+package org.wso2.carbon.connector.operations;
+
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.storage.file.datalake.DataLakeFileClient;
+import com.azure.storage.file.datalake.DataLakeFileSystemClient;
+import com.azure.storage.file.datalake.DataLakeServiceClient;
+import com.azure.storage.file.datalake.models.DataLakeStorageException;
+import com.azure.storage.file.datalake.models.LeaseAction;
+import com.azure.storage.file.datalake.options.DataLakeFileAppendOptions;
+import org.apache.synapse.MessageContext;
+import org.json.JSONObject;
+import org.wso2.carbon.connector.connection.AzureStorageConnectionHandler;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.core.connection.ConnectionHandler;
+import org.wso2.carbon.connector.util.AbstractAzureMediator;
+import org.wso2.carbon.connector.util.AzureConstants;
+import org.wso2.carbon.connector.util.Error;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implements the append file operation.
+ */
+public class AppendFile extends AbstractAzureMediator {
+
+    @Override
+    public void execute(MessageContext messageContext, String responseVariable, Boolean overwriteBody) {
+
+        String connectionName = getProperty(messageContext, AzureConstants.CONNECTION_NAME, String.class, false);
+        String fileSystemName =
+                getMediatorParameter(messageContext, AzureConstants.FILE_SYSTEM_NAME, String.class, false);
+        String filePathToAppend =
+                getMediatorParameter(messageContext, AzureConstants.FILE_PATH_TO_APPEND, String.class, false);
+        String inputType = getMediatorParameter(messageContext, AzureConstants.INPUT_TYPE, String.class, false);
+        String localFilePath =
+                getMediatorParameter(messageContext, AzureConstants.LOCAL_FILE_PATH, String.class,
+                        !inputType.equals(AzureConstants.L_LOCAL_FILE_PATH));
```

Comment:
Possible NPE here. Let's update the code as below.

`(AzureConstants.L_LOCAL_FILE_PATH).equals(inputType)`